pkg/operator/controller/garden/access/reconciler.go

```diff
@@ -0,0 +1,140 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package access
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/afero"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kubernetesclient "github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operator/apis/config"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	"github.com/gardener/gardener/pkg/operator/controller/gardenlet"
+)
+
+// Reconciler reconciles garden access secrets.
+type Reconciler struct {
+	Client  client.Client
+	Manager manager.Manager
+	Config  *config.OperatorConfiguration
+
+	FS            afero.Fs
+	tokenFilePath string
+
+	GardenletControllerAdded bool
```

Comment:
```suggestion
	Config  *config.OperatorConfiguration
	FS            afero.Fs

	tokenFilePath string
	gardenletControllerAdded bool
```

Reply:
`gardenletControllerAdded` is exposed for the integration test where we actually skip adding the Gardenlet controller.

Reply:
See comments below which might make this field obsolete anyhow.